src/app/backend/resource/common/pod.go

```diff
@@ -79,6 +79,9 @@ func GetContainerNames(podTemplate *v1.PodSpec) []string {
 	for _, container := range podTemplate.Containers {
 		containerNames = append(containerNames, container.Name)
 	}
+	for _, container := range podTemplate.InitContainers {
```

Comment:
Hmm...

Currently, GetContainerNames() is almost identical to get GetContainerImages(). Also, it is currently simple logic in common package. From that reason I would not merge init container and normal containers in this method + package.

Currently, GetContainerNames() is only used by logs. If you make this method more log specific you should move the method to logs package during your refactoring, I guess.

Reply:
Thanks @cheld, I think that's valid.

I noticed the `GetContainerImages` method in common does almost exactly the same thing as `GetContainerNames` (but for images, not names, obviously); the Container Images code will also have the same issue as this PR, that init containers aren't included. Is this something which should be corrected? I'm less familiar with where `GetContainerImages` is used, but presumably if it's presenting a list of Images to a user it would be reasonable to expect init containers to be included too.

With that in mind, an alternative refactoring could be merging `GetContainerNames` and `GetContainerImages` into a single method returning a `ContainerInfo{Name,Image,Type string}` struct. Thoughts?

I'm easy either way; it just seems we have an option of pulling these methods further apart, or pushing them together.

Reply:
Yesterday I had the feeling that we have no problem, because we show init containers in the UI. Now, thinking again about it, I think e.g. controller list views like ReplicaSet-List-View will probably miss init containers. 
It would be fantastic if you search for usage of the method and check the UI if we have problems. You can create separate issue for it.

If we have the same issue in GetContainerNames and GetContanerImages I would merge. If not I would rather move appart.

Fundamentally, your proposal is fine. But I guess it is not working - but maybe I am wrong. The goal of the method was to reduce a little bit of boiler plate code in the caller. I think with your proposal we have to iterate in the caller again. So there seems no benefit. We just convert one struct in another :)) . I tink adding parameters would work